.github/workflows/ios-tests.yml

```diff
@@ -21,11 +21,17 @@ jobs:
                 sudo xcode-select -s /Applications/Xcode_15.3.app/Contents/Developer
                 gem install xcpretty
             - name: Build
-              run: xcodebuild build-for-testing -workspace apps/ios/GuideDogs.xcworkspace -scheme Soundscape -destination 'platform=iOS Simulator,name=iPhone 15' CODE_SIGN_IDENTITY= CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO |tee xcodebuild.log | xcpretty
+              run: >
+                xcodebuild build-for-testing -workspace apps/ios/GuideDogs.xcworkspace
+                -scheme Soundscape -destination 'platform=iOS Simulator,name=iPhone 15,OS=18.0'
+                CODE_SIGN_IDENTITY= CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO
+                | tee xcodebuild.log | xcpretty && exit ${PIPESTATUS[0]}
```

Comment:
Rather than hard-code the iOS version here, ChatGPT suggests you can query the latest stable iOS version by adding a stage like this:
```
    - name: Find latest stable iOS version
      id: ios-version
      run: |
        # Get the latest stable iOS runtime version
        LATEST_IOS_VERSION=$(xcrun simctl list runtimes | grep -Eo 'iOS [0-9]+\.[0-9]+' | sort -r | grep -v 'beta' | head -n 1 | awk '{print $2}')
        echo "Latest iOS version: $LATEST_IOS_VERSION"
        echo "LATEST_IOS_VERSION=$LATEST_IOS_VERSION" >> $GITHUB_ENV
```
And then you'd use the argument `-destination 'platform=iOS Simulator,name=iPhone 15,OS=$LATEST_IOS_VERSION'` here and below.